docs/guides/latest/auto-backup/workload.md

```diff
@@ -0,0 +1,719 @@
+```
+
+## Backup Deployment
+
+Now, we are going to backup a Deployment using the template we have configured earlier. We will mount two ConfigMap volume in two different directories of the Deployment. Then, we will backup those directories using automatic backup.
+
+**Create Deployment:**
+
+Below is the YAML of the Deployment and respective ConfigMaps that we are going to create,
+
+```yaml
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: stash-sample-data-1
+  namespace: demo
+data:
+  file1.txt: "Data from ConfigMap 'stash-sample-data-1'"
+---
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: stash-sample-data-2
+  namespace: demo
+data:
+  file2.txt: "Data from ConfigMap 'stash-sample-data-2'"
+---
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  labels:
+    app: stash-demo
+  name: stash-demo
+  namespace: demo
+  annotations:
+    stash.appscode.com/backup-template: workload-backup-template
+    stash.appscode.com/target-directories: "/source/data-1,/source/data-2"
+    stash.appscode.com/volume-mounts: "source-data-1:/source/data-1,source-data-2:/source/data-2"
+spec:
+  replicas: 3
+  selector:
+    matchLabels:
+      app: stash-demo
+  template:
+    metadata:
+      labels:
+        app: stash-demo
+      name: busybox
+    spec:
+      containers:
+      - args:
+        - sleep
+        - "3600"
+        image: busybox
+        imagePullPolicy: IfNotPresent
+        name: busybox
+        volumeMounts:
+        - mountPath: /source/data-1
+          name: source-data-1
+        - mountPath: /source/data-2
+          name: source-data-2
+      restartPolicy: Always
+      volumes:
+      - name: source-data-1
+        configMap:
+          name: stash-sample-data-1
+      - name: source-data-2
+        configMap:
+          name: stash-sample-data-2
+```
+
+Notice the `metadata.annotations` field. We have specified the automatic backup specific annotations to backup `/source/data-1` and `/source/data-2` directories of the `source-data-1` and `source-data-2` volumes respectively. We have also specified to use `workload-backup-template` BackupConfigurationTemplate for creating `Repository` and `BackupConfiguration` for this Deployment.
+
+Let's create the above Deployment,
+
+```console
+$ kubectl apply -f ./docs/examples/guides/latest/auto-backup/workload/deployment.yaml
+configmap/stash-sample-data-1 created
+configmap/stash-sample-data-2 created
+deployment.apps/stash-demo created
+```
+
+If everything goes well, Stash will create a `Repository` and a `BackupConfiguration` with the name in the following format: `<workload-kind>-<workload-name>`.
+
+**Verify Repository:**
+
+Verify that the Repository has been created successfully by the following command,
+
+```console
+$ kubectl get repository -n demo
+NAME                    INTEGRITY   SIZE   SNAPSHOT-COUNT   LAST-SUCCESSFUL-BACKUP   AGE
+deployment-stash-demo                                                                9s
```

Comment:
Should we change the format to `${TARGET_KIND}`-`${TARGET_NAMESPACE}`-`${TARGET_NAME}`

Reply:
Ignore this, since repository and backupconfiguration are both namespaced.